File: 1400-construct-k-palindrome-strings/1400-construct-k-palindrome-strings.java
class Solution {
    public boolean canConstruct(String s, int k) {
        if(s.length() < k)
            return false;
        int[] charCount = new int['z' - 'a' + 1];
        int odd = 0;
        for(char c : s.toCharArray())charCount['z' - c]++;
        for(int cC: charCount){
            if(cC > 0)
                odd += cC % 2;
        }
        return odd <= k ? true : false;
    }
}

File: 1400-construct-k-palindrome-strings/1400-construct-k-palindrome-strings.py
class Solution:
    def canConstruct(self, s: str, k: int) -> bool:
        
        if len(s) < k:
            return False
        
        hashmap = {}
        
        for x in s:
            if x not in hashmap:
                hashmap[x] = 1
            else:
                hashmap[x] += 1
        odd = 0
        for key, values in hashmap.items():
            if values %2 != 0:
                odd += 1
        if odd <= k:
            return True
        else: return False
            
        